src/os/bluestore/bluefs_types.h

```diff
@@ -71,14 +130,19 @@ struct bluefs_fnode_t {
 
   uint64_t allocated;
   uint64_t allocated_commited;
+  uint8_t type = 0; // was prefer_bdev
+  uint64_t wal_limit; // EOF of wal, this limit represents upper limit of fnode.size, not upper limit of wal_size
+  uint64_t wal_size; // Size we expect is in WAL, there could be more on power off instances in range of wal_size~wal_limit
```

Comment:
I would phrase it:
"Without envelopes. Size of data in WAL, there ... range of fnode.size~wal_limit"